### ISSUE TITLE
CatalogBrowser: check on init whether agreement exists

### ISSUE THREAD

**Issue by paullatzelsperger** [agera-edc/EDCDataDashboard#11]
The `CatalogBrowserComponent` should check during initialization whether a contract exists for each offer and act as follows:
- if exists: 
  - re-label the button to `"Re-Negotiate"`
  - add a link to navigate to the agreements page (depends #9)
- if not exists keep as is

**Comment by zeier** [comment]
@paullatzelsperger are you still working on this issue?

**Comment by paullatzelsperger** [comment]
@zeier yes, and eventually, i'll complete it. not very high prio as of now, so it may sit on the back burner for some time.
